### PR TITLE
Tests. Add test for helper_api::helper_get_link_attributes

### PR DESCRIPTION
**Pull request by grummbeer**
This is the first commit of a series with the attempt to modernize the tests.

Issue: [#34379](https://mantisbt.org/bugs/view.php?id=34379) 

```
vendor/bin/phpunit tests/Mantis/Helper/GetLinkAttributesTest.php
PHPUnit 9.6.18 by Sebastian Bergmann and contributors.

.......                                                             7 / 7 (100%)
```

### CODE REVIEW

File: tests/Mantis/Helper/GetLinkAttributesTest.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
```
Comment (dregad) [inline tests/Mantis/Helper/GetLinkAttributesTest.php:3]: This should be moved below the comment blocks (before the require_once call).
And follow coding guidelines.

File: tests/Mantis/Helper/GetLinkAttributesTest.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+/**
+ * Test for helper_api::helper_get_link_attributes
+ *
+ * @see helper_get_link_attributes()
+ */
+class GetLinkAttributesTest extends MantisCoreBase
+{
+	/**
+	 * Link attributes are correctly created accordingly to the
```
Comment (dregad) [inline tests/Mantis/Helper/GetLinkAttributesTest.php:38]: `according to`, not <del>accordingly to</del>

File: tests/Mantis/Helper/GetLinkAttributesTest.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		config_set('html_make_links', $p_config );
+
+		# 1.
+		$this->assertSame( $p_array, helper_get_link_attributes() );
+		# 2.
+		$this->assertSame( $p_string, trim( helper_get_link_attributes( false ) ) );
+	}
+
+	public function provideConfigurations(): \Generator {
+		yield 'LINKS_NEW_WINDOW' => [
```
Comment (dregad) [inline tests/Mantis/Helper/GetLinkAttributesTest.php:57]: I have actually never used this language construct before... Nice 😃

**Review by dregad — CHANGES_REQUESTED** [review]
Thanks for this.

I would suggest to document the test data sets' structure in PHPDoc, either with @param in testHelperReturnsArrayOrString block, or in provideConfigurations.

Commit message should reference the bugtracker issue it fixes (i.e. `Fixes #xxx` or `Issue #xxx` depending on whether you just want to automatically resolve the issue, or just reference it), so it gets automatically picked up by the source integration plugin.

File: tests/Mantis/Helper/GetLinkAttributesTest.php (hunk 2 above)
Comment (grummbeer) [reply tests/Mantis/Helper/GetLinkAttributesTest.php]: fixed.

Thank you, please - you know my bad English - proofreading is really necessary.

File: tests/Mantis/Helper/GetLinkAttributesTest.php (hunk 1 above)
Comment (grummbeer) [reply tests/Mantis/Helper/GetLinkAttributesTest.php]: Moved.

> And follow coding guidelines.

Are there any other objections?

File: tests/Mantis/Helper/GetLinkAttributesTest.php (hunk 3 above)
Comment (grummbeer) [reply tests/Mantis/Helper/GetLinkAttributesTest.php:56]: No need for creating an array.

**Comment by grummbeer** [comment]
Fixed the array style. ready.

File: tests/Mantis/Helper/GetLinkAttributesTest.php (hunk 2 above)
Comment (dregad) [reply tests/Mantis/Helper/GetLinkAttributesTest.php]: Your English is really not that bad 😃

**Review by dregad — APPROVED** [review]

**Pull request merged**